src/ngraph_encapsulate_op.cc

```diff
@@ -229,9 +251,15 @@ class NGraphEncapsulateOp : public tf::OpKernel {
 
       // Create the nGraph output tensor
       void* dst_ptr = tf::DMAHelper::base(output_tensor);
-      auto t_result = m_ng_backend->create_tensor(elem_type, shape, dst_ptr);
+      if(env_value == "NNP") {
+      	auto t_result = m_ng_backend->create_tensor(elem_type, shape);
+        outputs.push_back(t_result);
+      }
+      else {
+      	auto t_result = m_ng_backend->create_tensor(elem_type, shape, dst_ptr);
+        outputs.push_back(t_result);
+      }
```

Comment:
There are missing parts:
1) before `m_ng_backend->call()`, do a tensor copy from host->device
2) after `m_ng_backend->call()`, do a tensor copy from device->host
Otherwise, it doesn't seem that there is a place where NNP is receiving the data.

Reply:
Same on GPU.